src/java.desktop/share/classes/sun/swing/FilePane.java

```diff
@@ -1216,6 +1251,15 @@ public Component getTableCellRendererComponent(JTable table, Object value,
 
             return this;
         }
+
+        private void updateMessageFormatPattern(String pattern) {
+            mf.applyPattern(pattern);
+            mf.setFormat(0, nf);
+        }
+
+        private static double formatToDoubleValue(long len) {
+            return (len / 100L) / 10.0d;
```

Comment:
First "formatToDouble" isn't really a very good name for this. 
Perhaps I could provide a better one if I could be sure of what the intent is here but
at the very least this method needs comments explaining what it is doing AND why.

The first (len / 100L) expression means that (eg)
100 and 199 will both become 1 and the return value of the method is 0.1 

So rounding DOWN is happening here, whereas (separately) we round UP for value < 100 ..

So anything from 1  byte to 199 bytes will be 0.1 KB ?

The test doesn't seem to cover this case.

Reply:
"formatToDouble" method is taking filesize as a long value in parameter and convert it to one decimal precision value. 
The reason for converting to one decimal point is to make it similar to native file system which shows file size in one decimal precision.

can "formatToDouble" method rename to "getOneDecimalPrecisionFileSize" ?

Since we are displaying file size in one decimal precision, files having size from 1 to 199 byte will be 0.1 KB. 
1 to 99 byte file size is handled separately otherwise it will show 0.0 KB.

I will add few test case for 100 to 199 byte file.

Reply:
> Since we are displaying file size in one decimal precision, files having size from 1 to 199 byte will be 0.1 KB.

Yes, I see that. My point is it seems wrong.

Reply:
Can you please tell me what should I do to handle 1 byte to199 bytes ?